docs/intro-react.md

```diff
@@ -0,0 +1,476 @@
+
+First, you will want to import `useState` from React like so:
+
+```jsx
+import React, {useState} from 'react';
+```
+
+In this example, inside your `Cat` function, `useState` creates a `name` state variable:
+
+```jsx
+const [name, setName] = useState("What's my name again?");
+```
+
+Then you add `onChangeText` to the `<TextInput>`. This fires every time the `<TextInput>`‘s value changes (every time a user adds a character of text).
+
+```jsx
+onChangeText={(text) => setName(text)}
+```
+
+This uses the `setName` you defined with the `useState` Hook above to update your state variable `defaultText` every time `onChangeText` fires. You might set state when you have new data from the server, or from user input, or even from a timer.
+
+> You might’ve noticed that although `name` is a [const](https://developer.mozilla.org/Web/JavaScript/Reference/Statements/const), it is seemingly reassignable! What is happening is that every time a user inputs a letter into `<TextInput>`, `setName()` fires to update the state const `name`. When a state-setting function like `setName` is called, its component will re-render, running its function again. In this case `Cat` will re-render its `<Cat>`—discarding the old state const `name` in the process!
+
+<block class="classical syntax" />
+The older class components approach is a little different when it comes to state.
+
+```SnackPlayer name=State%20and%20Class%20Components
+import React, { Component } from 'react';
+import { Text, TextInput, View } from 'react-native';
+
+export class Cat extends Component {
+  state = {name: "What's my name again?!"}
+
+  render(){
+    return (
+      <View>
+        <Text>Hello, I am...{this.state.name}!</Text>
+        <TextInput
+          style={{
+            height: 40,
+            borderColor: 'gray',
+            borderWidth: 1
+          }}
+          onChangeText={
+            (text) => this.setState({name: text})
+          }
+        />
+      </View>
+    );
+  }
+}
+
+export default class Cafe extends Component {
+  render(){
+    return (
+      <>
+        <Cat />
+        <Cat />
+      </>
+    );
+  }
+}
+```
+
+In class components, state is stored in a `state` object:
+
+```jsx
+state = {name: "What's my name again?!"};
+```
+
+You access this object inside your component with `this.state`:
+
+```jsx
+<Text>Hello, I am...{this.state.name}!</Text>
+```
+
+And you set individual values inside the state object with `this.setState()`:
+
+```jsx
+this.setState({name: text});
+```
+
+> Do not change your component's state directly by assigning it a new value with `this.state.name = text`. Calling `this.setState()` allows React to track changes made to state that trigger rerendering. Setting state directly can break your app's reactivity!
+
+<block class="endBlock syntax" />
+
+---
```

Comment:
In general, we might want to rethink this example. First, because you have the next chapter about TextInput anyway — but we use it earlier. Second, because text input involves explaining "controlled components" — like inputs with `value` prop. But most people find it counter-intuitive at first. What's the point of state if the input works "on its own" anyway? They expect input to remember its own state.

So an example that isn't an input might work better here. So that it's obvious *why* the component wants to "remember" something. This also ensures people who don't yet understand controlled components wouldn't struggle with state in general. State is simple.

Reply:
In fact, "Handling Text Inputs" should probably explain the concept of a controlled component instead. And `value` vs `defaultValue`. It's ok if that's out of scope of this PR though.

Reply:
Handling Text Inputs is out of the scope of this PR, as it's slated to be scooped up into a minitutorial. 

Trying to think of other things that we can use state for. I want to keep UI cruft to a minimum—which is why I reached for inputText. And no to timers—too contrived, and personally I know I and other trip on timers. Let me ponder.

Reply:
I showed you the Cat Cafe and milk demo. I'm going to go with that one—you press on a button to feed the cats at the cat cafe.